views/index.ejs

```diff
@@ -8,10 +8,13 @@
     <h1>Shopping for Cookies</h1>
 
 
-    <strong>Items in cart: <!-- Print the number of items in the cart here --></strong>
+    <strong>Items in cart: <%= cart.length; %> <!-- Print the number of items in the cart here --></strong>
     <ul>
       <!-- List the names of the items in your cart here -->
+     <% cart.forEach(function(product){ %>
+       <li><%= product.name %></li>
```

Comment:
You're on the right track with printing the name, but we'll also need the id in order to complete the post request with req.body.id in your route. Since displaying that id isn't really necessary, we can set it to the value of a hidden input.

You are also going to need a form here so you can submit the post request. We need that submit button!
```html
<% cart.forEach(function(product){ %>
       <form action="'/cart'" method="post">
            <li><%= product.name %></li>
            <input type="hidden" name="id" value="<%= product.id %>">
      </form>
<% }); %>
```